src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonPatternDetector.cs

```diff
@@ -0,0 +1,307 @@
+            var argumentNode = literalNode.Parent;
+            Debug.Assert(_syntaxFacts.IsArgument(argumentNode));
+
+            var argumentList = argumentNode.Parent;
+            var invocationOrCreation = argumentList.Parent;
+            if (_syntaxFacts.IsInvocationExpression(invocationOrCreation))
+            {
+                var invokedExpression = _syntaxFacts.GetExpressionOfInvocationExpression(invocationOrCreation);
+                var name = GetNameOfInvokedExpression(invokedExpression);
+                if (_syntaxFacts.StringComparer.Equals(name, _methodNameOfInterest))
+                {
+                    // Is a string argument to a method that looks like it could be a Regex method.  
+                    // Need to do deeper analysis
+                    var method = _semanticModel.GetSymbolInfo(invocationOrCreation, cancellationToken).GetAnySymbol();
+                    if (method != null &&
+                        method.DeclaredAccessibility == Accessibility.Public &&
+                        method.IsStatic &&
+                        _typesOfInterest.Contains(method.ContainingType))
+                    {
+                        return AnalyzeStringLiteral(
+                            stringLiteral, argumentNode, cancellationToken);
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public JsonTree TryParseJson(SyntaxToken token, CancellationToken cancellationToken)
+        {
+            if (IsDefinitelyNotJson(token, _syntaxFacts))
+            {
+                return null;
+            }
+
+            HasJsonLanguageComment(token, _syntaxFacts, out var strict);
+
+            var chars = _virtualCharService.TryConvertToVirtualChars(token);
+            if (chars.IsDefaultOrEmpty)
+            {
+                return null;
+            }
+
+            return JsonParser.TryParse(chars, strict);
+        }
+
+        private bool AnalyzeStringLiteral(
+            SyntaxToken stringLiteral, SyntaxNode argumentNode,
+            CancellationToken cancellationToken)
+        {
+            var parameter = _semanticFacts.FindParameterForArgument(_semanticModel, argumentNode, cancellationToken);
+            return parameter?.Name == _jsonName;
+        }
+
+        private string GetNameOfInvokedExpression(SyntaxNode invokedExpression)
+        {
+            if (_syntaxFacts.IsSimpleMemberAccessExpression(invokedExpression))
+            {
+                return _syntaxFacts.GetIdentifierOfSimpleName(_syntaxFacts.GetNameOfMemberAccessExpression(invokedExpression)).ValueText;
+            }
+            else if (_syntaxFacts.IsIdentifierName(invokedExpression))
+            {
+                return _syntaxFacts.GetIdentifierOfSimpleName(invokedExpression).ValueText;
+            }
+
+            return null;
+        }
+
+
+        public bool IsProbablyJson(SyntaxToken token, CancellationToken cancellationToken)
```

Comment:
This method doesn't seem to be used.

Reply:
It is used in: AbstractJsonStringDetectorDiagnosticAnalyzer

There we check if it looks like it's probably json, and doesn't have any markers that indicate it's definitely json (i.e. it's **not** "JToken.Parse").  In that case, we offer to add the ```// lang=json``` comment to indicate that we should light-up language services here.